plugin/pkg/admission/persistentvolumeclaim/default/admission_test.go

```diff
@@ -0,0 +1,232 @@
+		TypeMeta: unversioned.TypeMeta{
+			Kind: "PersistentVolumeClaim",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Name:      "claimWithEmptyClass",
+			Namespace: "ns",
+			Annotations: map[string]string{
+				classAnnotation: "",
+			},
+		},
+	}
+	claimWithNoClass := &api.PersistentVolumeClaim{
+		TypeMeta: unversioned.TypeMeta{
+			Kind: "PersistentVolumeClaim",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Name:      "claimWithNoClass",
+			Namespace: "ns",
+		},
+	}
+
+	tests := []struct {
+		name              string
+		classes           []*extensions.StorageClass
+		claim             *api.PersistentVolumeClaim
+		expectError       bool
+		expectedClassName string
+	}{
+		{
+			"no default, no modification of PVCs",
+			[]*extensions.StorageClass{classWithFalseDefault, classWithNoDefault, classWithEmptyDefault},
+			claimWithNoClass,
+			false,
+			"",
+		},
+		{
+			"one default, modify PVC with class=nil",
+			[]*extensions.StorageClass{defaultClass1, classWithFalseDefault, classWithNoDefault, classWithEmptyDefault},
+			claimWithNoClass,
+			false,
+			"default1",
+		},
+		{
+			"one default, no modification of PVC with class=''",
+			[]*extensions.StorageClass{defaultClass1, classWithFalseDefault, classWithNoDefault, classWithEmptyDefault},
+			claimWithEmptyClass,
+			false,
+			"",
+		},
+		{
+			"one default, no modification of PVC with class='foo'",
+			[]*extensions.StorageClass{defaultClass1, classWithFalseDefault, classWithNoDefault, classWithEmptyDefault},
+			claimWithClass,
+			false,
+			"foo",
+		},
+		{
+			"two defaults, error with PVC with class=nil",
+			[]*extensions.StorageClass{defaultClass1, defaultClass2, classWithFalseDefault, classWithNoDefault, classWithEmptyDefault},
+			claimWithNoClass,
+			true,
+			"",
+		},
+		{
+			"two defaults, no modification of PVC with class=''",
+			[]*extensions.StorageClass{defaultClass1, defaultClass2, classWithFalseDefault, classWithNoDefault, classWithEmptyDefault},
+			claimWithEmptyClass,
+			false,
+			"",
+		},
+		{
+			"two defaults, no modification of PVC with class='foo'",
+			[]*extensions.StorageClass{defaultClass1, defaultClass2, classWithFalseDefault, classWithNoDefault, classWithEmptyDefault},
+			claimWithClass,
+			false,
+			"foo",
+		},
+	}
+
+	for _, test := range tests {
+		glog.V(4).Infof("starting test %q", test.name)
+
+		// clone the claim, it's going to be modified
+		clone, err := conversion.NewCloner().DeepCopy(test.claim)
+		if err != nil {
+			t.Fatalf("Cannot clone claim: %v", err)
+		}
+		claim := clone.(*api.PersistentVolumeClaim)
+
+		ctrl := newPlugin(nil)
+		for _, c := range test.classes {
+			ctrl.store.Add(c)
+		}
+		attrs := admission.NewAttributesRecord(
+			claim, // new object
+			nil,   // old object
+			api.Kind("PersistentVolumeClaim").WithVersion("version"),
+			claim.Namespace,
+			claim.Name,
+			api.Resource("persistentvolumeclaims").WithVersion("version"),
+			"", // subresource
+			admission.Create,
```

Comment:
if you want to have the behavior also applied on updates, add a test for update as well.